File: poc/scripts/create_master_and_cards.js
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

const PRIVATE_DIR = path.join(__dirname, '..', 'private');
if (!fs.existsSync(PRIVATE_DIR)) fs.mkdirSync(PRIVATE_DIR, { recursive: true });
const MASTER_KEY_PATH = path.join(PRIVATE_DIR, 'master.key');
const CARDS_ENC_PATH = path.join(PRIVATE_DIR, 'cards.enc');

const cards = [
  { id: 1, title: 'Silent Move', text: 'Move quietly, keep conversations to hand signals.' },
  { id: 2, title: 'Buddy Check', text: 'Regularly check buddy gear and breathing.' },
  { id: 3, title: 'Route Selection', text: 'Choose paths that minimize exposure and maximize cover.' },
  { id: 4, title: 'Signal', text: 'Use compact signals to convey intent without noise.' },
  { id: 5, title: 'Watch Rotation', text: 'Rotate watch duties to maintain alertness.' }
];

function encryptWithKey(plainBuffer, key) {
  const iv = crypto.randomBytes(12);
  const cipher = crypto.createCipheriv('aes-256-gcm', key, iv);
  const ct = Buffer.concat([cipher.update(plainBuffer), cipher.final()]);
  const tag = cipher.getAuthTag();
  return Buffer.concat([iv, ct, tag]).toString('base64');
}

function main() {
  const newMaster = crypto.randomBytes(32);
  fs.writeFileSync(MASTER_KEY_PATH, newMaster.toString('base64'), { mode: 0o600 });
  console.log('Wrote new master.key');

  const cardsJson = JSON.stringify(cards);
  const enc = encryptWithKey(Buffer.from(cardsJson, 'utf8'), newMaster);
  fs.writeFileSync(CARDS_ENC_PATH, enc, { mode: 0o600 });
  console.log('Wrote new cards.enc');
}

main();

File: poc/scripts/create_master_and_cards.py
from cryptography.hazmat.primitives.ciphers.aead import AESGCM
import os, base64, json, secrets

PRIVATE_DIR = os.path.join(os.getcwd(), 'private')
os.makedirs(PRIVATE_DIR, exist_ok=True)

MASTER_KEY_PATH = os.path.join(PRIVATE_DIR, 'master.key')
CARDS_ENC_PATH = os.path.join(PRIVATE_DIR, 'cards.enc')

cards = [
  {"id":1,"title":"Silent Move","text":"Move quietly, keep conversations to hand signals."},
  {"id":2,"title":"Buddy Check","text":"Regularly check buddy gear and breathing."},
  {"id":3,"title":"Route Selection","text":"Choose paths that minimize exposure and maximize cover."},
  {"id":4,"title":"Signal","text":"Use compact signals to convey intent without noise."},
  {"id":5,"title":"Watch Rotation","text":"Rotate watch duties to maintain alertness."}
]

def encrypt_with_key(plain_bytes, key):
    aesgcm = AESGCM(key)
    iv = secrets.token_bytes(12)
    ct = aesgcm.encrypt(iv, plain_bytes, None)
    return iv + ct

def main():
    new_master = secrets.token_bytes(32)
    with open(MASTER_KEY_PATH, 'wb') as f:
        f.write(base64.b64encode(new_master))
    print('WROTE_MASTER', MASTER_KEY_PATH)

    cards_json = json.dumps(cards).encode('utf8')
    combined = encrypt_with_key(cards_json, new_master)
    with open(CARDS_ENC_PATH, 'wb') as f:
        f.write(base64.b64encode(combined))
    print('WROTE_CARDS', CARDS_ENC_PATH)

if __name__ == '__main__':
    main()
